e2e/machine_migration_test.go

```diff
@@ -506,3 +683,71 @@ func cleanupMachineResources(ctx context.Context, cl client.Client, capiMachines
 		mapiframework.WaitForMachinesDeleted(cl, m)
 	}
 }
+func updateAWSMachineProviderSpec(machine *machinev1beta1.Machine, updateFunc func(*machinev1beta1.AWSMachineProviderConfig)) error {
+	providerSpec, err := getAWSProviderSpecFromMachine(machine)
+	if err != nil {
+		return err
+	}
+
+	updateFunc(providerSpec)
+
+	rawProviderSpec, err := json.Marshal(providerSpec)
+	if err != nil {
+		return fmt.Errorf("failed to marshal provider spec: %w", err)
+	}
+
+	machine.Spec.ProviderSpec.Value.Raw = rawProviderSpec
+	return nil
+}
+func verifyUpdatePrevented(cl client.Client, machineName string, description string, updateFunc func(*machinev1beta1.Machine), expectedErrorSubstrings ...string) {
+	By(fmt.Sprintf("Attempting to %s in MAPI Machine", description))
+
+	currentMachine, err := mapiframework.GetMachine(cl, machineName)
+	Expect(err).NotTo(HaveOccurred(), "Failed to get current machine")
+	updatedMachine := currentMachine.DeepCopy()
+	updateFunc(updatedMachine)
```

Comment:
I think all of this could be just passing the updatefunc and machine to a `k.Update`, and we just pass the machine in question (rather than always passing machine.Name)

```go
					Eventually(k.Update(machine, updateFunc), timeout).Should(MatchError(ContainSubstring(expectedErrorSubstrings)))
```

and we move the logic for working out what `expectedErrorSubstrings` should be to above the call.

Reply:
Let me add this .